src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,66 @@
+import { useContext, useState } from "react";
+import { UsernameContext } from "../contexts/Username";
+import { postCommentByUsername } from "../api";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCheckCircle } from "@fortawesome/free-solid-svg-icons";
+
+function CommentAdder({ article_id }) {
+	const { username } = useContext(UsernameContext);
+	const [commentInput, setCommentInput] = useState("");
+	const [isPosting, setIsPosting] = useState(false);
+	const [error, setError] = useState("");
+	const [postSuccess, setPostSuccess] = useState(false);
+
+	function handleChange(event) {
+		setError("");
+		setCommentInput(event.target.value);
+	}
+
+	function handleClick(event) {
+		event.preventDefault();
+		setIsPosting(true);
+		postCommentByUsername(article_id, username, commentInput)
+			.then((response) => {
+				setCommentInput("");
+				setPostSuccess(true);
+				setTimeout(() => setPostSuccess(false), 4000);
+				setTimeout(() => setIsPosting(false), 4000);
+			})
+			.catch((err) => {
+				setIsPosting(false);
+				setError(err.response.data.msg);
+			});
+	}
+
+	return (
```

Comment:
Consider adding updating the dependency of the fetching comments useEffect so that it updates the list when the comment has successfully been posted or optimistically updating the comments list when the comment is posted as at the moment I have to refresh the page in order to see the new comment i have added even though i see visual feedback which tells me the comment has been added! :)